test/suites/smoke/test-old-regressions.ts

```diff
@@ -0,0 +1,198 @@
+          },
+        ],
+        name: "balanceOf",
+        outputs: [
+          {
+            internalType: "uint256",
+            name: "",
+            type: "uint256",
+          },
+        ],
+        stateMutability: "view",
+        type: "function",
+      },
+    ],
+    functionName: "balanceOf",
+    args: ["0x30763be2bf075c3fDeA704c5f59A76d011d02943", 2n],
+  }),
+};
+
+// MOON-2822
+const moon2822: BadBlockRegressionCase = {
+  issue: "MOON-2822",
+  network: Network.Moonbeam,
+  contractAddress: "0xFfFFfFff1FcaCBd218EDc0EbA20Fc2308C778080",
+  block: "latest",
+  callData: encodeFunctionData({
+    abi: [
+      {
+        inputs: [],
+        name: "totalSupply",
+        outputs: [],
+        stateMutability: "nonpayable",
+        type: "function",
+      },
+    ],
+    functionName: "totalSupply",
+    args: [],
+  }),
+};
+
+const cases = [moon2824, moon2822];
+
+describeSuite({
+  id: "S15",
+  title: "Verify regressions which happened in the past by reading historical state",
+  foundationMethods: "read_only",
+  testCases: async ({ context, it, log }) => {
+    let paraApi: ApiPromise;
+
+    beforeAll(async function () {
+      paraApi = context.polkadotJs("para");
+      const chainId = (await paraApi.query.ethereumChainId.chainId()).toString();
+      log(`Loading test data for chainId ${chainId}.`);
+    });
+
+    it({
+      id: "C001",
+      title: "Verify all bad block regression cases",
+      test: async function () {
+        for (const testCase of cases) {
+          if (testCase.network != (paraApi.consts.system.version.specName.toString() as Network)) {
+            log(`Skipping... (Issue ${testCase.issue} specific for ${testCase.network})`);
+            continue;
+          }
+
+          const callParams = {
+            to: testCase.contractAddress,
+            data: testCase.callData,
+          };
+          // Add either blockTag or blockNumber depending on the case specification
+          if (typeof testCase.block === "string") {
+            callParams["blockTag"] = testCase.block;
+          } else if (typeof testCase.block === "bigint") {
+            callParams["blockNumber"] = testCase.block;
+          }
+
+          const result = await context.viem().call(callParams);
+
+          try {
+            expect(result.data).to.contain("0x");
+          } catch (e) {
+            log(
+              `Error found for ${testCase.issue} at block ${testCase.block.toString()}: ${
+                result.data
+              }`
+            );
+            throw e;
+          }
+        }
+      },
+    });
+
+    it({
+      id: "C002",
+      title: "Verify bad transaction tracing case",
+      chainType: "moonbeam",
+      test: async function () {
+        // Fetch and verify the trace of a bad transaction observed in client version 0.38
+        // Detailed in MOON-2702
+        const badTxHash = "0xd91d98b539720d8a42069268126d366fd29165e487d94b165a97e0158842657b";
```

Comment:
what about MOON-2936?

We want to trace one transaction per runtime, not just this one.

Reply:
Added test case 003 to check for a TX for every runtime